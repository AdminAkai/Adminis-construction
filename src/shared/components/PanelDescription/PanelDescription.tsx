import { FC, PropsWithChildren } from 'react'

import styles from './panelDescription.module.css'

const PanelDescription: FC<PropsWithChildren> = ({ children }) => (
  <p className={styles['panel-description']} aria-label='Project Description'>
    {children}
  </p>
)

export default PanelDescription
