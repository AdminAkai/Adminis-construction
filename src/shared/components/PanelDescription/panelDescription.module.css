.project-description {
  margin-top: 0.5rem;
  color: var(--text-secondary);
  font-size: 1.5rem;
}

@media screen and (max-width: 959px) {
  .project-description {
    font-size: 1rem;
  }
}