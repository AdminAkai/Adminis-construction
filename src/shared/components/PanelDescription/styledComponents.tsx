import styled from 'styled-components'

export const ProjectDescription = styled.p`
  margin-top: 0.5rem;
  color: ${({ theme }) => theme.secondary};
  font-size: 1.5rem;

  @media screen and (max-width: ${({ theme }) => theme.mobileMediaQuery}) {
    font-size: 1rem;
  }
`
