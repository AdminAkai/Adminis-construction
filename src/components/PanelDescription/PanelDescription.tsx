import { FC, PropsWithChildren } from 'react'

import { ProjectDescription } from './styledComponents'

const PanelDescription: FC<PropsWithChildren> = ({ children }) => (
  <ProjectDescription aria-label='Project Description'>
    {children}
  </ProjectDescription>
)

export default PanelDescription
