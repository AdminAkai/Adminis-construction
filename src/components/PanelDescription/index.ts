import PanelDescription from './PanelDescription'

export default PanelDescription
