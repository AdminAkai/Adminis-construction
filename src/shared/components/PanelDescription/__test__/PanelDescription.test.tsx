import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'

import PanelDescription from '../PanelDescription.tsx'

const testDescription: string = 'test description'

describe('PanelDescription tests', () => {
  it('Make sure PanelDescription renders children', () => {
    render(<PanelDescription>{testDescription}</PanelDescription>)

    const panelDescription = screen.getByLabelText('Project Description')

    expect(panelDescription).toBeVisible()
    expect(panelDescription).toHaveTextContent(testDescription)
  })

  it('PanelDescription snapshot renders correctly', () => {
    const tree = renderer
      .create(<PanelDescription>{testDescription}</PanelDescription>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
