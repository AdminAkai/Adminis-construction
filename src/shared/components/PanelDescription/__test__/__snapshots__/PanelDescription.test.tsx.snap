// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelDescription tests PanelDescription snapshot renders correctly 1`] = `
<p
  aria-label="Project Description"
  className="sc-braxZu jhDjdk"
>
  test description
</p>
`;
